### PR TITLE
Escape Newline/Paragraph separators

### PR DESCRIPTION
**Pull request by ajhyndman**
I think that the Webpack JSON loader seems like the right place to handle 
this issue.

For whatever reason, the JSON and Javascript specifications disagree on
whether or not strings can contain the unicode Newline or Paragraph 
characters.

In the case that a JSON object containing one or more of these characters is
being printed to a script tag, we should escape them.

See also this discussion:
https://github.com/expressjs/express/issues/1132#issuecomment-10100419

### CODE REVIEW

**Comment by laughinghan** [comment]
See also: http://timelessrepo.com/json-isnt-a-javascript-subset

**Comment by ajhyndman** [comment]
Demo of a failing import:
http://www.webpackbin.com/N1L-cuOAg

Click on "Log" at the top left to see the error generated.
Delete or escape the character "\u2028" from illegal.json
to see the expected behaviour.

**Comment by graingert** [comment]
@ajhyndman you should rebase and sign the CLA

**Comment by laughinghan** [comment]
Christ. @ajhyndman: I'm happy to do it if you give me push access to your fork

File: index.js [hunk 1]
```diff
@@ -6,5 +6,9 @@ module.exports = function(source) {
 	this.cacheable && this.cacheable();
 	var value = typeof source === "string" ? JSON.parse(source) : source;
 	this.value = [value];
-	return "module.exports = " + JSON.stringify(value, undefined, "\t") + ";";
+	return "module.exports = " +
+			JSON.stringify(value, undefined, "\t")
+					.replace(/\u2028/g, '\\u2028')
+					.replace(/\u2029/g, '\\u2029') +
```
Comment (graingert) [inline index.js]: you don't need regexes for this:
.replace('\u2028', '\\u2028')
.replace('\u2029', '\\u2029')

Comment (sokra) [inline index.js]: replace with string replaces only once. You need the regexp with g flag.

Comment (laughinghan) [inline index.js]: oh my goodness how did I not know that, that's really surprising

**Comment by ajhyndman** [comment]
@laughinghan Sure, thanks!

**Comment by laughinghan** [comment]
@ajhyndman: you gotta give me push access to your fork though

File: index.js (hunk 1 above)
Comment (graingert) [inline index.js]: sorry, thought I was in Python land

**Comment by laughinghan** [comment]
@graingert @sokra Rebased!

**Comment by michael-ciniawsky** [comment]
@ajhyndman Can you please close and reopen the PR to trigger the CLA Bot again ? 😛

**Comment by laughinghan** [comment]
@ajhyndman: I don't think you need to do that, just go to https://cla.js.foundation/webpack-contrib/json-loader?pullRequest=18 and sign it and come back and I'd expect it to update

**Comment by graingert** [comment]
@bebraw Sure, but there are no existing tests here.  it's unreasonable to expect a contributor to write the whole testing infrastructure.

**Comment by bebraw** [comment]
@graingert Ah, yeah. Fair point. Two options:

1. We'll accept the PR now without tests.
2. We'll accept the PR with tests after integrating **webpack-defaults** to the projects. That gives the testing infrastructure needed.

**Comment by michael-ciniawsky** [comment]
@bebraw I'm working on various webpack-defaults upgrades for the 'smaller' loaders, but it will take a few days, it's not best pratice by any means but @graingert has a point here, since there isn't any test setup provided for contribs, we can always revert this change if the issue tracker blows up but thats my personal opinion (I don't think so) 😛

**Comment by bebraw** [comment]
@michael-ciniawsky Yeah, it's a simple change so merge now. Easy enough to add a good test later.

**Comment by michael-ciniawsky** [comment]
@ajhyndman Please close and reopen the PR to trigger the CLA Bot again , otherwise I can't merge this for legislative reasons 😛

**Comment by michael-ciniawsky** [comment]
@ajhyndman ping 😛

**Comment by ajhyndman** [comment]
Sorry!  Signed, and fixed up merge conflicts.

**Pull request merged**